File: Oline-AQ-spring/src/main/java/com/example/olineaqspring/entity/Exam.java
package com.example.olineaqspring.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("exam")
public class Exam {
    @TableId(type = IdType.AUTO)
    private Integer examId;
    private String examName;
    private String description;
    private Integer duration;
    private BigDecimal totalScore;
    private String status;
    private Boolean allowRetake;
    private Boolean assignAll;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime createTime;
}

File: Oline-AQ-app/app/src/main/java/com/onlineaq/student/data/model/Exam.kt
package com.onlineaq.student.data.model

data class Exam(
    val examId: Int,
    val examName: String,
    val description: String?,
    val duration: Int,
    val totalScore: Double?,
    val status: String,
    val allowRetake: Boolean?,
    val assignAll: Boolean?,
    val startTime: String?,
    val endTime: String?,
    val createTime: String?
)

data class ExamDetail(
    val exam: Exam,
    val questions: List<Question>
)
